src/components/Form.js

```diff
@@ -239,9 +255,14 @@ export default class Form extends Component {
       newFormData = this.getUsedFormData(this.state.formData, fieldNames);
     }
 
+    let errors = [];
+    let errorSchema = {};
     if (!this.props.noValidate) {
-      const { errors, errorSchema } = this.validate(newFormData);
-      if (Object.keys(errors).length > 0) {
+      const validationResult = this.validate(newFormData);
+      errors = validationResult.errors;
```

Comment:
i tried doing this with a deconstructor, but got a boolean is not a function error, very strange.
let canSubmit
({errrors, errorSchema, canSubmit} = this.validate(newFormData)